universe/marathon.json.mustache

```diff
@@ -3,38 +3,45 @@
     "cpus": {{service.cpus}},
     "mem": {{service.mem}},
     "cmd": "/sbin/init.sh",
+    {{#service.user}}
     "user": "{{service.user}}",
+    {{/service.user}}
+
     "env": {
-{{#security.kerberos.enabled}}
-        {{#security.kerberos.krb5conf}}
+        {{#security.kerberos.enabled}}
+            {{#security.kerberos.krb5conf}}
             "SPARK_MESOS_KRB5_CONF_BASE64": "{{security.kerberos.krb5conf}}",
-        {{/security.kerberos.krb5conf}}
-        {{#security.kerberos.kdc.hostname}}
+            {{/security.kerberos.krb5conf}}
+            {{#security.kerberos.kdc.hostname}}
             "SPARK_SECURITY_KERBEROS_KDC_HOSTNAME": "{{security.kerberos.kdc.hostname}}",
-        {{/security.kerberos.kdc.hostname}}
-        {{#security.kerberos.kdc.port}}
+            {{/security.kerberos.kdc.hostname}}
+            {{#security.kerberos.kdc.port}}
             "SPARK_SECURITY_KERBEROS_KDC_PORT": "{{security.kerberos.kdc.port}}",
-        {{/security.kerberos.kdc.port}}
-        {{#security.kerberos.realm}}
+            {{/security.kerberos.kdc.port}}
+            {{#security.kerberos.realm}}
             "SPARK_SECURITY_KERBEROS_REALM": "{{security.kerberos.realm}}",
-        {{/security.kerberos.realm}}
-{{/security.kerberos.enabled}}
+            {{/security.kerberos.realm}}
+        {{/security.kerberos.enabled}}
 
-{{#service.use_bootstrap_for_IP_detect}}
-{{/service.use_bootstrap_for_IP_detect}}
-{{^service.use_bootstrap_for_IP_detect}}
-        "SKIP_BOOTSTRAP_IP_DETECT": "true",
-{{/service.use_bootstrap_for_IP_detect}}
+        {{^service.virtual_network_enabled}}
```

Comment:
I wonder if this might be causing the Json issue? At first I thought you were missing the opening `#service.virtual_network_enabled` but I see you put it below on L31. I think the the json string needs to be closed or the extra `,` on L22 might be messing things up in the test

nit: can you also fix it the structure back to `if/else` style?

Reply:
Changed the order of checks to follow `if/else` style. The final JSON seems to be generated properly - let's wait for CI tests to pass.